extensions/ql-vscode/test/vscode-tests/minimal-workspace/queries-panel/query-discovery.test.ts

```diff
@@ -16,31 +19,63 @@ describe("QueryDiscovery", () => {
     expect(workspace.workspaceFolders?.length).toEqual(1);
   });
 
+  function mockResolveQueryByLanguage(
+    dataFn: (queryDir: string) => {
+      byLanguage?: Record<string, string[]>;
+      noDeclaredLanguage?: string[];
+      multipleDeclaredLanguages?: string[];
+    },
+  ) {
+    return jest.fn().mockImplementation((queryDir: Uri) => {
+      const data = dataFn(queryDir.fsPath);
+      const value: QueryInfoByLanguage = {
+        byLanguage: Object.keys(data.byLanguage || {}).reduce((result, key) => {
```

Comment:
Could we use `Object.entries` here instead of `Object.keys` to avoid needing to call `data.byLanguage![key]`?